File: codes/cpp/wk01_heat_modes.cpp
// Wk01 - Conduction vs convection vs radiation (CSV output)
// Build: g++ -O2 -std=c++17 wk01_heat_modes.cpp -o modes && ./modes
// Output: heat_modes.csv (Ts, q_cond, q_conv, q_rad in W/m^2)
// Author: Prof. S. Joon Kwon - SPMDL - SKKU
#include <cstdio>
#include <cmath>

int main() {
    const double SIGMA = 5.670e-8, T_INF = 25.0;
    const double K = 0.6, L = 0.02, H = 25.0, EPS = 0.85;

    FILE* f = std::fopen("heat_modes.csv", "w");
    std::fprintf(f, "Ts_C,q_cond,q_conv,q_rad\n");
    double cross = -1.0, prev = -1.0;
    for (int i = 0; i <= 400; ++i) {
        double Ts = 30.0 + (900.0 - 30.0) * i / 400.0;
        double dT = Ts - T_INF;
        double qc = K * dT / L;
        double qv = H * dT;
        double qr = EPS * SIGMA *
            (std::pow(Ts + 273.15, 4) - std::pow(T_INF + 273.15, 4));
        std::fprintf(f, "%.1f,%.1f,%.1f,%.1f\n", Ts, qc, qv, qr);
        double d = qr - qv;
        if (prev < 0 && d >= 0 && cross < 0) cross = Ts;
        prev = d;
    }
    std::fclose(f);
    std::printf("Radiation passes convection near Ts = %.0f C\n", cross);
    std::puts("Wrote heat_modes.csv");
    return 0;
}

File: codes/matlab/wk01_heat_modes.m
% Wk01 - Conduction vs convection vs radiation (T^4 takeover)
% Author: Prof. S. Joon Kwon - SPMDL - SKKU
clear; clc;

SIGMA = 5.670e-8; T_inf = 25;
k = 0.6; L = 0.02; h = 25; eps = 0.85;

Ts = linspace(30, 900, 400);
dT = Ts - T_inf;
q_cond = k * dT / L;
q_conv = h * dT;
q_rad  = eps * SIGMA * ((Ts+273.15).^4 - (T_inf+273.15)^4);

figure(1);
plot(Ts, q_cond/1e3, 'LineWidth', 2); hold on;
plot(Ts, q_conv/1e3, 'LineWidth', 2);
plot(Ts, q_rad/1e3,  'LineWidth', 2); hold off;
grid on;
xlabel('hot-surface temperature T_s [C]');
ylabel('heat flux q'''' [kW/m^2]');
title('Three modes of heat transfer: who dominates when?');
legend('conduction k\DeltaT/L','convection h\DeltaT', ...
       'radiation \epsilon\sigma(T_s^4-T_\infty^4)', ...
       'Location','northwest');

[~, ic] = min(abs(q_rad - q_conv));
fprintf('Radiation passes convection near Ts = %.0f C\n', Ts(ic));
